src/vs/platform/terminal/common/capabilities/shellEnvDetectionCapability.ts

```diff
@@ -11,7 +11,8 @@ import { equals } from '../../../../base/common/objects.js';
 export class ShellEnvDetectionCapability extends Disposable implements IShellEnvDetectionCapability {
 	readonly type = TerminalCapability.ShellEnvDetection;
 
-	private readonly _env: Map<string, string> = new Map();
+	private _env: Map<string, string> = new Map();
+	private _pendingEnv: Map<string, string> | undefined;
 	get env(): Map<string, string> { return this._env; }
```

Comment:
So private and public env are together:

```suggestion
	private _pendingEnv: Map<string, string> | undefined;
	private _env: Map<string, string> = new Map();
	get env(): Map<string, string> { return this._env; }
```